### PR TITLE
feat: add transaction dialog in saving plugin

### PR DESCRIPTION
**Pull request by nuanyang233**
## Description

Closes MF-293

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/theme/src/Components/SearchableList/SearchableList.tsx [hunk 1]
```diff
@@ -72,7 +72,7 @@ export function SearchableList<T extends {}>({
                 shouldSort: true,
                 threshold: 0.45,
                 minMatchCharLength: 1,
-                keys: searchKey ?? data.length > 0 ? Object.keys(data[0]) : [],
+                keys: searchKey ?? Object.keys(data.length > 0 ? data[0] : []),
```
Comment (guanbinrui) [inline packages/theme/src/Components/SearchableList/SearchableList.tsx:75]:
```suggestion
                keys: searchKey ?? data.length > 0 ? Object.keys(data[0]) : [],
```

Comment (guanbinrui) [inline packages/theme/src/Components/SearchableList/SearchableList.tsx:75]: So, they are different, right?

Comment (nuanyang233) [reply packages/theme/src/Components/SearchableList/SearchableList.tsx:75]: Yes, they will cause #6073

Comment (UncleBill) [reply packages/theme/src/Components/SearchableList/SearchableList.tsx:75]: I see, it's about the operator precedence.

**Review by BillySh1 — APPROVED** [review]

File: packages/mask/src/plugins/Savings/SNSAdaptor/SavingsForm.tsx [hunk 2]
```diff
@@ -41,6 +47,11 @@ export interface SavingsFormProps {
     onClose?: () => void
 }
 
+const ProtocolName = {
+    [ProtocolType.Lido]: 'lido',
+    [ProtocolType.AAVE]: 'aave',
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Savings/SNSAdaptor/SavingsForm.tsx:52]:
```suggestion
    [ProtocolType.Lido]: 'Lido',
    [ProtocolType.AAVE]: 'AAVE',
```

Comment (guanbinrui) [reply packages/mask/src/plugins/Savings/SNSAdaptor/SavingsForm.tsx:52]: Recommend to use `createLookupTableResolver`, you will never lost new `ProtocolType`s.

File: packages/mask/src/plugins/Savings/SNSAdaptor/SavingsForm.tsx [hunk 3]
```diff
@@ -50,6 +61,10 @@ export function SavingsForm({ chainId, protocol, tab, onClose }: SavingsFormProp
     const { NATIVE_TOKEN_ADDRESS } = useTokenConstants()
     const [inputAmount, setInputAmount] = useState('')
     const [estimatedGas, setEstimatedGas] = useState<BigNumber.Value>(new BigNumber('0'))
+    const [tradeState, setTradeState] = useState<TransactionState>({
+        type: TransactionStateType.UNKNOWN,
+    })
+    const [isOpen, setIsOpen] = useState(false)
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Savings/SNSAdaptor/SavingsForm.tsx:67]:
```suggestion
    const [open, setOpen] = useState(false)
```

**Pull request merged**